app/controllers/members_controller.rb

```diff
@@ -162,8 +165,14 @@ def self.tab_scripts
     @@scripts.join('(); ') + '();'
   end
 
+  def set_roles_and_principles!
+    @roles = Role.find_all_givable
+    # Check if there is at least one principal that can be added to the project
+    @principals_available = @project.possible_members("", 1)
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.